src/integration/utils/fetchDom.js

```diff
@@ -22,8 +22,10 @@ const faultTolerantDomFetch = (url, runScripts) =>
       }
 
       try {
-        const dom = await JSDOM.fromURL(
-          url,
+        const response = await fetch(url, headers && { headers });
```

Comment:
The `fromUrl` method doesn't allow to pass in headers, so this has been replaced with a regular `fetch` to be able to do so. Then we use the default `new JSDOM` and pass in the HTML to create a DOM fragment.